### PR TITLE
refactor: remove `data_hash` from CellOutput

### PR DESCRIPTION
**Pull request by quake**
This PR removed `data_hash` from CellOutput and refactored `Transaction` struct in `ckb.mol`

### CODE REVIEW

File: script/src/verify.rs [hunk 1]
```diff
@@ -552,7 +553,7 @@ mod tests {
             runner: Runner::default(),
         };
         let verifier = TransactionScriptsVerifier::new(&rtx, &data_loader, &config);
-
+        verifier.verify(100).unwrap();
```
Comment (u2) [inline script/src/verify.rs]: remove this line?

Comment (quake) [reply script/src/verify.rs]: ops, added for debug, forgot to delete...

File: script/src/syscalls/mod.rs [hunk 2]
```diff
@@ -425,8 +425,8 @@ mod tests {
 
     proptest! {
         #[test]
-        fn test_load_cell_partial(ref data in any_with::<Vec<u8>>(size_range(1000).lift())) {
-            _test_load_cell_partial(data)?;
+        fn test_load_cell_partial(ref data in any_with::<Vec<u8>>(size_range(1000).lift()), offset in any::<u64>()) {
```
Comment (jjyr) [inline script/src/syscalls/mod.rs]: We should indicate a range for offset before we can rely on `proptest` default behavior.

File: script/src/syscalls/mod.rs [hunk 3]
```diff
@@ -43,7 +43,9 @@ pub const DEBUG_PRINT_SYSCALL_NUMBER: u64 = 2177;
 #[derive(Debug, PartialEq, Clone, Copy, Eq)]
 enum CellField {
     Capacity = 0,
-    DataHash = 1,
+    // just comment out for compatibility
+    // https://github.com/nervosnetwork/ckb-system-scripts/blob/aa5583b21129bf5b4d66d9a34262295c852ac6fe/c/ckb_consts.h#L29
+    // DataHash = 1,
```
[the remark this answers is not in the document]
Comment (quake) [reply script/src/syscalls/mod.rs]: let's upgrade it in a separate PR.

File: script/src/syscalls/mod.rs (hunk 2 above)
Comment (quake) [reply script/src/syscalls/mod.rs]: Updated, please review again, thanks.

File: miner/src/block_assembler.rs [hunk 4]
```diff
@@ -485,10 +485,7 @@ impl BlockAssembler {
                 data.truncate(data_max_len);
             }
 
-            let output_new = output
-                .as_builder()
-                .data_hash(CellOutput::calc_data_hash(&data).pack())
-                .build();
+            let output_new = output.as_builder().build();
```
[the remark this answers is not in the document]
Comment (quake) [reply miner/src/block_assembler.rs]: Updated, please review again, thanks.

File: test/src/node.rs [hunk 5]
```diff
@@ -365,15 +365,13 @@ impl Node {
...
-            .to_entity()
-            .data_hash()
-            .to_owned()
-            .unpack();
+        self.always_success_code_hash = CellOutput::calc_data_hash(
+            &consensus.genesis_block().transactions()[0]
+                .outputs_data()
+                .get(1)
+                .unwrap()
+                .raw_data(),
```
Comment (u2) [inline test/src/node.rs]: :+1:

File: util/types/schemas/ckb.mol [hunk 6]
```diff
@@ -72,18 +71,14 @@ table RawTransaction {
     header_deps:    Byte32Vec,
     inputs:         CellInputVec,
     outputs:        CellOutputVec,
+    outputs_data:   BytesVec,
```
Comment (u2) [inline util/types/schemas/ckb.mol]: :+1:

**Comment by u2** [comment]
Please update the commit message and description, this commit includes the refactor of `Transaction` structure.

**Comment by u2** [comment]
ci: 

> error[E0599]: no method named `data_hash` found for type `ckb_types::generated::types::CellOutputBuilder` in the current scope

**Comment by quake** [comment]
> Please update the commit message and description, this commit includes the refactor of `Transaction` structure.

Update the PR

**Comment by quake** [comment]
> ci:
> 
> > error[E0599]: no method named `data_hash` found for type `ckb_types::generated::types::CellOutputBuilder` in the current scope

rebased, please review again

**Comment by quake** [comment]
> Please update the commit message and description, this commit includes the refactor of `Transaction` structure.

updated, thanks.

**Comment by u2** [comment]
bors r=u2,jjyr,keroro520,yangby-cryptape

**Comment by u2** [comment]
bors delegate+

File: test/src/specs/tx_pool/different_txs_with_same_input.rs [hunk 7]
```diff
@@ -38,8 +38,8 @@ impl Spec for DifferentTxsWithSameInput {
         node0.generate_block();
         node0.generate_block();
 
-        info!("RBF (Replace-By-Fees) is not implemented");
```
Comment (u2) [inline test/src/specs/tx_pool/different_txs_with_same_input.rs]: :+1:

**Comment by quake** [comment]
bo

**Pull request merged**